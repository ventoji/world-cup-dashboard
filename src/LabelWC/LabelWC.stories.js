import React from 'react';
import LabelWC from './LabelWC';
import { withKnobs, text } from '@storybook/addon-knobs';

export default {
  title: 'WC Components/LabelWC',
  component: LabelWC,
  decorators: [withKnobs],
};

const Template = (args) => <LabelWC {...args} />;

export const Default = Template.bind({});

Default.args = {
  label: text('label', 'home Team'),
};
