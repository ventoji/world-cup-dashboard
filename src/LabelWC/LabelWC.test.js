import React from 'react';
import LabelWC from './LabelWC';
import { shallow } from 'enzyme';

describe('Label Component', () => {
  let label;

  beforeEach(() => {
    label = shallow(<LabelWC htmlFor="labelElement" label="field label" />);
  });

  it('Should be defined', () => {
    expect(LabelWC).toBeDefined();
  });

  it('Should render properly', () => {
    const element = label.find('label');

    expect(label).toMatchSnapshot();
    // label exists
    expect(element).toHaveLength(1);
    // label has attribute htmlFor
    expect(element.props().htmlFor).toMatch(/labelElement/);
    // label is the excpeted string defined as an input
    expect(element.props().children[0]).toEqual('field label');
  });

  it('Should render label with a red asterisk when field is required', () => {
    label = shallow(
      <LabelWC htmlFor="labelElement" label="field label" required />,
    );

    expect(label).toMatchSnapshot();
    const element = label.find('label span');
    expect(element.props().style).toEqual({ color: 'red' });
    expect(element.props().children).toMatch(/\*/);
  });
});
