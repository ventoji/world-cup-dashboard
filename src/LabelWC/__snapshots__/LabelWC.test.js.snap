// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component Should render label with a red asterisk when field is required 1`] = `ShallowWrapper {}`;

exports[`Label Component Should render properly 1`] = `ShallowWrapper {}`;
